### DISCUSSION TITLE
Add custom field to AdminSummary and to the admin default filter

### DISCUSSION THREAD

**Discussion by SamueleFurnari** [OrchardCMS/OrchardCore#10039] [Q&A] [answered]
Hello everybody!

I've created a new custom field.
Now I want to:

- show it's value in the Summary Admin as a simple text
- add an extra filter in the default search bar in the _Manage Content_ page, and manage it. 

For the first thing, I've entered in the field placement definition for summary views and I've used the following definition.

```
{
    "place": "Meta",
    "displayType": "SummaryAdmin",
    "differentiator": "test-Ufficio",
    "alternates": null,
    "wrappers": null,
    "shape": null
  }
```

The result is as reported in the following image:

![image](https://user-images.githubusercontent.com/6505846/128435848-ace6ac79-6a00-4830-9870-0df351cfc943.png)

I want to represent it as simple string and I think that I must use a template, but I don't find the correct help that solve my requirement.

For the additional filter I read the code but I don't see any point in wich I can insert my custom filter.

Any help is appreciated.

Thanks
Samuele

**Comment by SamueleFurnari** [comment, marked as answer]
For the first part of my question, I solved by creating a razor page with the name that respect the rules reported in templates documentation (the _[ShapeType].[DisplayType].cshtml_ rule, and so my file name is _MyField.SummaryAdmin.cshtml_).

I don't understand why this does not work if I create a template from the admin with the name _MyField_SummaryAdmin_.

**Reply by tropcicstefan** [reply]
You should initialize shape in display driver of field, or better yet of encompassing part. Something like in [PublishLaterPartDisplayDriver](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.PublishLater/Drivers/PublishLaterPartDisplayDriver.cs).

And for the filter you could implement IContentsAdminListFilter (like in [taxonomy](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Taxonomies/Services/TaxonomyContentsAdminListFilter.cs)) based on your field's values

**Reply by SamueleFurnari** [reply]
Great, thank you very much for the tip!
